contracts/libraries/MultiSend.sol

```diff
@@ -64,8 +64,9 @@ contract MultiSend {
                 }
                 if eq(success, 0) {
                     let errorLength := returndatasize()
-                    returndatacopy(0, 0, errorLength)
-                    revert(0, errorLength)
+                    let returnPtr := mload(0x40)
+                    returndatacopy(returnPtr, 0, errorLength)
+                    revert(returnPtr, errorLength)
```

Comment:
Nice! THis is a case where the assembly was marked as `memory-safe`, but it actually wasn't.

This fix should definitely be included in the PR.

Reply:
Just to be sure, this should be reverted back according to this comment?

https://github.com/safe-global/safe-contracts/pull/711/files#r1445800424
> In that case, I'm all for **not** changing the `MultiSend*` and `DelegateCaller` contracts.

Reply:
nit - for consistency, should we just call this `ptr` here as well?

Reply:
Based on my previous message about your comment @nlordell, this will be reverted. Just confirming because of your last message in this thread.

Reply:
Oh yes - this should be reverted - I missed what file it was part of.

I just noticed - but this *shouldn't* use a stack variable for `errorLength`: `returndatasize` is a 1-byte 2-gas opcode (the cheapest class of opcode) and is more efficient to use than storing it in a stack variable (which involves using `DUPn` which is 3-gas AFAIK).

This is **unrelated to this PR** however, but can be fixed in a follow up.

Reply:
Will also remove the `memory-safe-assembly` tagging just to be on the safe side. Will update the use of `returndatasize` as well 👍🏾

Reply:
Can you update the use of `returndatasize` in a separate PR since it is an unrelated change to this one?

Reply:
Will make a new PR once this is merged, made the revert 👍🏾